### PR TITLE
Support generation config in ORTModel

### PR DESCRIPTION
**Pull request by fxmarty**
This PR adds support for generation config in ORTModel, following https://github.com/huggingface/transformers/pull/20388

Note: we should really add nightly tests tracking on transformers/diffusers main.

### CODE REVIEW

**Comment by regisss** [comment]
Big +1 for nightly tests aligned with Transformers/Diffusers main branches!

So this PR cannot be merged before the next release of Transformers right?

**Comment by fxmarty** [comment]
Yes, I think we will need to wait for the next release of transformers to merge!

File: optimum/onnxruntime/modeling_seq2seq.py [hunk 1]
```diff
@@ -1076,13 +1083,28 @@ def _from_pretrained(
...
+                model_id,
+                cache_dir=cache_dir,
+                force_download=force_download,
+                local_files_only=local_files_only,
+                use_auth_token=use_auth_token,
+                revision=revision,
+                subfolder=subfolder,
+            )
+        except OSError:
+            logger.info("Generation config file not found, using a generation config created from the model config.")
```
Comment (mht-sharma) [inline optimum/onnxruntime/modeling_seq2seq.py:1098]:
```suggestion
            logger.warning("Generation config file not found, using a generation config created from the model config.")
```

File: optimum/onnxruntime/modeling_seq2seq.py [hunk 2]
```diff
@@ -1286,6 +1308,10 @@ def _reorder_cache(past, beam_idx) -> Tuple[Tuple[torch.FloatTensor]]:
             )
         return reordered_past
 
+    def can_generate(self):
```
Comment (mht-sharma) [inline optimum/onnxruntime/modeling_seq2seq.py:1311]: Could this be put in the `ORTModelForConditionalGeneration`?

File: optimum/onnxruntime/modeling_decoder.py [hunk 3]
```diff
@@ -626,13 +634,28 @@ def _from_pretrained(
...
+                model_id,
+                cache_dir=cache_dir,
+                force_download=force_download,
+                local_files_only=local_files_only,
+                use_auth_token=use_auth_token,
+                revision=revision,
+                subfolder=subfolder,
+            )
+        except OSError:
+            logger.info("Generation config file not found, using a generation config created from the model config.")
```
Comment (mht-sharma) [inline optimum/onnxruntime/modeling_decoder.py:649]:
```suggestion
            logger.warning("Generation config file not found, using a generation config created from the model config.")
```

File: optimum/onnxruntime/modeling_seq2seq.py (hunk 1 above)
Comment (fxmarty) [reply optimum/onnxruntime/modeling_seq2seq.py:1098]: I would prefer to leave as is as it follows https://github.com/huggingface/transformers/blob/375801d5e65457d2ef86e6b269703c5f9ad67252/src/transformers/modeling_utils.py#L2520-L2522

File: optimum/onnxruntime/modeling_decoder.py (hunk 3 above)
Comment (fxmarty) [reply optimum/onnxruntime/modeling_decoder.py:649]: Same as above

File: optimum/onnxruntime/modeling_seq2seq.py (hunk 2 above)
Comment (fxmarty) [reply optimum/onnxruntime/modeling_seq2seq.py:1317]: done

**Comment by fxmarty** [comment]
I'll merge  despite the red CI for exporters, that is a minor regression https://github.com/huggingface/optimum/issues/721

**Pull request merged**